.github/workflows/ci.yaml

```diff
@@ -77,3 +77,29 @@ jobs:
 
       - name: Check for unstaged files
         run: git diff --exit-code
+  fmt:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
```suggestion
```
This will wail the workflow if we run it manually using `workflow_dispatch`.
From docs:
>  The branch, tag or SHA to checkout. When checking out the repository that
   triggered a workflow, this defaults to the reference or SHA for that event.
   Otherwise, uses the default branch.
   ref: ''